File: lib/domain/shared/gateways/api_key_principal_gateway.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Gateways
      # API キー文字列から {SessionPrincipal} を返す（永続はアダプタのみ）。
      module ApiKeyPrincipalGateway
        # @param api_key [String]
        # @return [Domain::Shared::Dtos::SessionPrincipal, nil]
        def principal_for_api_key(api_key)
          raise NotImplementedError, "#{self.class}#principal_for_api_key"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/gateways/api_key_principal_gateway.rs
use crate::shared::dtos::SessionPrincipal;

/// Ruby: `Domain::Shared::Gateways::ApiKeyPrincipalGateway`
pub trait ApiKeyPrincipalGateway: Send + Sync {
    /// Ruby: `#principal_for_api_key(api_key)` — `SessionPrincipal` or nil
    fn principal_for_api_key(&self, api_key: &str) -> Option<SessionPrincipal>;
}
